arcgis-ios-sdk-samples/Analysis/Analyze hotspots/AnalyzeHotspotsViewController.swift

```diff
@@ -96,11 +92,14 @@ class AnalyzeHotspotsViewController: UIViewController, HotspotSettingsVCDelegate
                 self.mapView.map?.operationalLayers.add(result!.mapImageLayer!)
                 
                 //set map view's viewpoint to the new layer's full extent
-                (self.mapView.map?.operationalLayers.firstObject as! AGSLayer).load { (error: Error?) in
-                    if error == nil {
+                (self.mapView.map?.operationalLayers.firstObject as! AGSLayer).load { [weak self] (error: Error?) in
+                    guard let self = self else { return }
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
                         //set viewpoint as the extent of the mapImageLayer
                         if let extent = result?.mapImageLayer?.fullExtent {
-                            self.mapView.setViewpointGeometry(extent, completion: nil)
+                            self.mapView.setViewpointGeometry(extent)
                         }
                     }
```

Comment:
A weak variable doesn't always need to be unwrapped with `guard`. If it is only ever accessed once for any given pathway, then it can just be unwrapped inline. But if it can be accessed repeatedly, then it should be unwrapped, since it is generally not good to repeatedly access a `weak` reference. So it would have been fine to use optional chaining both here.